calico-3.30.yaml

```diff
@@ -686,6 +688,189 @@ subpackages:
+        contents:
+          packages:
+            - wait-for-it
+      pipeline:
+        - name: Test binaries
+          runs: |
+            test "$(readlink /goldmane)" = "/usr/bin/goldmane"
+            test "$(readlink /health)" = "/usr/bin/health"
+            test "$(readlink /flowgen)" = "/usr/bin/flowgen"
+            test "$(readlink /stream)" = "/usr/bin/stream"
+        - name: "start daemon on localhost"
+          uses: test/daemon-check-output
+          with:
+            start: "/goldmane"
+            timeout: 30
+            expected_output: |
+              Health enabled
+              Starting flow collector
+              Listening on
+            post: |
+              wait-for-it -t 10 --strict localhost:8080 -- echo "Server is up"
+              wait-for-it -t 10 --strict localhost:443 -- echo "Server is up"
+
+  - name: "calico-whisker-${{vars.short-package-version}}"
+    description: "Tigera OSS-UI (Whisker)"
+    dependencies:
+      replaces:
+        - nginx-mainline-config
+      provides:
+        - calico-whisker=${{package.full-version}}
+      runtime:
+        - nginx-mainline
+        - coreutils
+        - dash-binsh
+        - sed
+        - libacl1
+        - libattr1
+        - libcap
+        - libcrypt1
+        - libcrypto3
+        - libpcre2-8-0
+        - libselinux
+        - libssl3
+        - ncurses
+        - zlib
+        - glibc
+        - merged-bin
+        - merged-usrsbin
+        - wolfi-baselayout
+    pipeline:
+      - working-directory: whisker
+        pipeline:
+          - name: Build the UI
+            runs: |
+              yarn install --frozen-lockfile
+              yarn build
+          - name: Install the files
+            runs: |
+              # https://github.com/projectcalico/calico/blob/master/whisker/docker-image/Dockerfile
+              mkdir -p "${{targets.contextdir}}"/etc/nginx/
+              mkdir -p "${{targets.contextdir}}"/etc/nginx/conf.d/
+              install -Dm755 docker-image/nginx-start.sh "${{targets.contextdir}}"/usr/bin/nginx-start.sh
+              install -Dm755 docker-image/nginx.conf ${{targets.contextdir}}/etc/nginx/
+              install -Dm755 docker-image/default.conf ${{targets.contextdir}}/etc/nginx/conf.d/
+              install -Dm755 LICENSE "${{targets.contextdir}}"/licenses/LICENSE
+              mkdir -p "${{targets.contextdir}}"/usr/share/nginx/html
+              cp -avR dist/* "${{targets.contextdir}}"/usr/share/nginx/html/
+    test:
+      environment:
+        contents:
+          packages:
+            - wait-for-it
+            - grep
+            - curl
+            - shadow
+      pipeline:
+        - name: Test important files
+          runs: |
+            #test "$(readlink /usr/sbin/nginx)" = "/usr/bin/nginx"
+            stat /usr/bin/nginx-start.sh
+            if [ -z "$(ls -A /usr/share/nginx/html/)" ]; then
+              echo "/usr/share/nginx/html/ is empty"
+              exit 1
+            fi
```

Comment:
Instead of (or in addition to) this, could we `curl` something from the running server later, to confirm that (a) the files were installed, _and_ (b) nginx is correctly configured to serve them?

Reply:
Yep, actually in the next step, we run `/usr/bin/nginx-start.sh` to do further check.

Reply:
We only check that the server comes up: I'm suggesting that we confirm it's serving up the content that we expect it to.

Reply:
Added `curl` test! https://github.com/wolfi-dev/os/pull/53969/commits/2d42f20eb7098eda4491000d491dcb755e990c7a